pkg/operator/k8sutil/node.go

```diff
@@ -230,19 +236,41 @@ func GetKubernetesNodesMatchingRookNodes(rookNodes []rookalpha.Node, clientset k
 
 // RookNodesMatchingKubernetesNodes returns only the given Rook nodes which have a corresponding
 // match in the list of Kubernetes nodes.
-func RookNodesMatchingKubernetesNodes(rookNodes []rookalpha.Node, kubernetesNodes []v1.Node) []rookalpha.Node {
+func RookNodesMatchingKubernetesNodes(rookStorage rookalpha.StorageScopeSpec, kubernetesNodes []v1.Node) []rookalpha.Node {
 	nodes := []rookalpha.Node{}
 	for _, kn := range kubernetesNodes {
-		for _, rn := range rookNodes {
+		for _, rn := range rookStorage.Nodes {
 			if rookNodeMatchesKubernetesNode(rn, kn) {
 				rn.Name = normalizeHostname(kn)
+
+				// modify the node Location if topology awareness enabled
+				if rookStorage.TopologyAware == true {
+					rn.Location = nodeTopologyLocation(kn, rn.Location)
+				}
+
 				nodes = append(nodes, rn)
 			}
 		}
 	}
 	return nodes
 }
 
+func nodeTopologyLocation(kubeNode v1.Node, location string) string {
+	nodeLabels := kubeNode.ObjectMeta.Labels
+	locations := []string{location}
+
+	for label := range nodeLabels {
```

Comment:
Can you add a comment here about what is being parsed? I see later in the unit test, but it would be helpful to have here.

Reply:
Done.